programs/token-metadata/program/src/state/metadata.rs

```diff
@@ -210,6 +203,32 @@ impl Metadata {
             _ => (),
         }
 
+        // Update Authority can update this section.
+        match &args {
+            UpdateArgs::V1 {
+                new_update_authority,
+                ..
+            }
+            | UpdateArgs::AsUpdateAuthorityV2 {
+                new_update_authority,
+                ..
+            } => {
+                if let Some(authority) = new_update_authority {
+                    self.update_authority = *authority;
+                }
+            }
+            _ => (),
+        }
```

Comment:
Should this section be combined with the section at the top of the function that says "Only the Update Authority can update this section"?

Reply:
Good point – done!